Automattic/wp-calypso
# Turn Flux Action Creators into Redux ones

Convert lib/themes/actions.js to Redux-style action creators
(using `redux-thunk` for async actions, and `redux-analytics`
middleware for actions involving analytics).

Note that the `themes-last-event` reducer and store are removed,
as the information they hold can be obtained from the `themes-list`
reducer, while the analytics functionality is now implemented in
lib/themes/middlewares.

Props @mcsf 

CC @ehg @seear 

To test:
- Restart Calypso.
- Check that the Theme Showcase (`calypso.localhost:3000/design`) still works as it should, including multi-site, single-site wpcom, and single-site Jetpack views.
- Don't forget to test analytics (Enter `localStorage.setItem( 'debug', 'calypso:analytics' );` in the browser console, and reload) -- themes search and activation should trigger corresponding analytics events. Check that the correct `search_term` is recorded. (Be aware of #391, though -- `search_term` isn't properly recorded for activation or purchase from multisite.)
